### PR TITLE
Goal preview re-renders on new comment input

### LINKED ISSUE #1737: Goal preview re-renders on new comment input
Looks like comment hook somehow affects the whole popup render dependency.

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes
- [x] Bug Fix

## Related issues
Resolve #1737 

## QA Instructions, Screenshots, Recordings

https://github.com/taskany-inc/issues/assets/95316053/94eb8955-34d9-4fea-9166-379fd2249eaf

### CODE REVIEW

**Review by awinogradov — APPROVED** [review]

File: src/components/GoalCommentCreateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import CommentCreateForm from './CommentCreateForm/CommentCreateForm';
+
+interface GoalCommentCreateFormProps {
+    states: ComponentProps<typeof CommentCreateForm>['states'];
+    goalId: string;
+    onSubmit: (comment?: { description: string; stateId?: string }) => Promise<Comment | null | undefined>;
+}
+
+const GoalCommentCreateForm: FC<GoalCommentCreateFormProps> = ({ states, goalId, onSubmit }) => {
```
Comment (KateKate) [inline src/components/GoalCommentCreateForm.tsx:14]: It's not critical, but I don't see the point in using FC.
```js
const GoalCommentCreateForm: FC<GoalCommentCreateFormProps> = ({ states, goalId, onSubmit }) => {
// =>
const GoalCommentCreateForm = ({ states, goalId, onSubmit }: GoalCommentCreateFormProps) => {
```

File: src/hooks/useLatest.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useInsertionEffect, useRef } from 'react';
+
+export function useLatest<T>(value: T) {
+    const ref = useRef(value);
+
+    useInsertionEffect(() => {
```
Comment (KateKate) [inline src/hooks/useLatest.ts:6]: Why do you use useInsertionEffect instead of useEffect/useLayoutEffect?

**Review by KateKate — COMMENTED** [review]
.

File: src/hooks/useLatest.ts (hunk 2 above)
Comment (DenisVorop) [reply src/hooks/useLatest.ts:6]: Callback refs are called before useLayoutEffect, which can lead to an irrelevant values in the callback refs.

useInsertionEffect is called before the callback refs, which guarantees the relevance of the values.

**Pull request merged**